### ISSUE TITLE
[Performance][VoxCPM2]: TTFA starts queueing at 2 RPS on H100

### ISSUE THREAD

**Issue by lolyhop** [vllm-project/vllm-omni#6365]
### Proposal to improve performance

Under steady open-loop load, VoxCPM2 starts reporting scheduler queue time at
2 RPS. One-second Prometheus samples at this level showed 1-2 running requests
and approximately one waiting request on average. Mean
`request_queue_time_seconds` was 391 ms, while audio TTFA p95 increased from
194 ms at 1 RPS to 842 ms.

Throughput stops scaling at about 5.4 requests/s. From 6 to 10 offered RPS,
mean server inference time remains close to 1.37 s, while mean queue time grows
from 1.85 s to 12.0 s.

We would like to understand what this queue time represents for VoxCPM2. In
particular, is it expected from decode-first scheduling, or is another
admission or model-specific constraint delaying new requests?

This may be related to #3874 / #3882 and #4176. The symptom is also similar to
#5365, but this test uses vLLM-Omni 0.26.0 and fixed-rate open-loop arrivals
rather than a single concurrent burst.

### Report of performance regression

This is not a version-to-version regression report. The measurements below are
a vLLM-Omni 0.26.0 baseline.

#### Setup

- one NVIDIA H100 PCIe 80 GB
- vLLM 0.26.0
- vLLM-Omni 0.26.0
- PyTorch 2.11.0+cu129
- FlashAttention 3
- 24 vCPU cgroup quota
- VoxCPM2-compatible fine-tuned checkpoint
- direct `/v1/audio/speech` request to the vLLM-Omni server; no reverse proxy

Server command (checkpoint path omitted):

```bash
vllm-omni serve <checkpoint> --omni \
  --host 0.0.0.0 \
  --port 8010 \
  --served-model-name voxcpm2 \
  --max-num-seqs 8 \
  --max-num-batched-tokens 4096 \
  --gpu-memory-utilization 0.9 \
  --max-model-len 4096 \
  --hf-overrides '{"voxcpm2_runtime_config":{"vae_decode_every":3,"audio_emit_every":1}}'
```

The remaining VoxCPM2 runtime options use the 0.26.0 dataclass defaults. This
includes batched CFM, batched VAE decode, coalesced audio D2H, fused LocDiT
QKV/MLP and the unified decode graph.

Request body:

```json
{
  "input": "<Russian text, 140-151 characters>",
  "response_format": "pcm",
  "stream": true,
  "stream_format": "audio"
}
```

The workload contains five Russian texts with lengths 140, 146, 151, 149 and
147 characters. Each RPS level sends requests at fixed intervals for 30
seconds. Sending is not gated on completion, and the client waits for all
requests to finish before starting the next level. The model and concurrency
shapes were warmed before the measured run.

#### Metrics

Audio TTFA is measured at the first complete 20 ms PCM16/48 kHz window with
RMS greater than 100. It is not based on HTTP headers or an SSE event. Through
5 RPS, the p95 difference between first response-body byte and verified audio
TTFA was 24-37 ms.

RTF is request wall time divided by generated audio duration. Completed req/s
uses the full level window, including queue drain.

Queue and inference time come from the server-side vLLM Prometheus histograms
`vllm:request_queue_time_seconds` (time in `WAITING`) and
`vllm:request_inference_time_seconds` (time in `RUNNING`). For each RPS level,
the mean is calculated from the counter deltas before the level and after its
queue has drained:

```text
mean = (histogram_sum_after - histogram_sum_before)
     / (histogram_count_after - histogram_count_before)
```

| offered RPS | audio TTFA p50 / p95 | RTF p50 / p95 | completed req/s | mean queue time | mean inference time |
|---:|---:|---:|---:|---:|---:|
| 1 | 115 / 194 ms | 0.098 / 0.105 | 1.00 | 0.001 s | 0.904 s |
| 2 | 433 / 842 ms | 0.132 / 0.179 | 1.96 | 0.391 s | 0.969 s |
| 3 | 745 / 1,227 ms | 0.176 / 0.236 | 2.90 | 0.523 s | 1.119 s |
| 4 | 847 / 1,445 ms | 0.202 / 0.268 | 3.80 | 0.600 s | 1.287 s |
| 5 | 881 / 1,512 ms | 0.213 / 0.281 | 4.71 | 0.617 s | 1.355 s |
| 6 | 2,152 / 3,509 ms | 0.350 / 0.488 | 5.29 | 1.852 s | 1.359 s |
| 7 | 4,723 / 8,023 ms | 0.602 / 0.962 | 5.32 | 4.244 s | 1.364 s |
| 8 | 7,054 / 13,117 ms | 0.849 / 1.487 | 5.42 | 6.613 s | 1.371 s |
| 9 | 9,764 / 17,906 ms | 1.138 / 1.963 | 5.41 | 9.417 s | 1.367 s |
| 10 | 12,588 / 23,103 ms | 1.624 / 2.535 | 5.39 | 12.007 s | 1.379 s |

Peak vLLM KV-cache usage stayed below 0.8%, and GPU memory allocated by the
process was about 13.3 GiB.

Questions:

1. At 2 RPS, one-second samples show only 1-2 running requests, but
   mean queue time is already 391 ms. Is this expected from decode-first
   scheduling, or is another admission/model-specific constraint involved?
2. What configuration should be used as the low-TTFA baseline on a single H100
   before profiling this further? Should `max_num_seqs` and
   `unified_decode_graph_max_batch_size` be raised together, or should the
   packaged `voxcpm2.yaml` be used unchanged?

### Your current environment (if you think it is necessary)

<details>
<summary>Relevant collect_env.py output</summary>

```text
OS                           : Ubuntu 22.04.4 LTS (x86_64)
GCC version                  : Ubuntu 11.4.0
Libc version                 : glibc-2.35
Python version               : 3.12.13
PyTorch version              : 2.11.0+cu129
CUDA used to build PyTorch   : 12.9
Is CUDA available            : True
GPU                          : NVIDIA H100 PCIe 80 GB
Nvidia driver version        : 580.65.06
CPU                          : Intel Xeon Gold 6338
CPU quota                    : 24 vCPU
vLLM Version                 : 0.26.0
vLLM-Omni Version            : 0.26.0
flashinfer-python            : 0.6.14
transformers                 : 5.14.1
triton                       : 3.6.0
```

</details>

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

**Comment by hsliuustc0106** [comment]
@linyueqian PTAL

**Comment by linyueqian** [comment]
ok

**Comment by lolyhop** [comment]
@linyueqian, take a look, please 🙏
